cob_obstacle_distance/include/cob_obstacle_distance/marker_shapes_impl.hpp

```diff
@@ -0,0 +1,142 @@
+/*
+ *****************************************************************
+ * \file
+ *
+ * \note
+ *   Copyright (c) 2015 \n
+ *   Fraunhofer Institute for Manufacturing Engineering
+ *   and Automation (IPA) \n\n
+ *
+ *****************************************************************
+ *
+ * \note
+ *   Project name: care-o-bot
+ * \note
+ *   ROS stack name: cob_control
+ * \note
+ *   ROS package name: cob_obstacle_distance
+ *
+ * \author
+ *   Author: Marco Bezzon, email: Marco.Bezzon@ipa.fraunhofer.de
+ *
+ * \date Date of creation: May, 2015
+ *
+ * \brief
+ *   This header contains the declaration and template implementation of MarkerShapes
+ *   which represent a combination of ROS markers and FCL geometric shapes.
+ *
+ ****************************************************************/
+
+#ifndef MARKER_SHAPES_IMPL_HPP_
+#define MARKER_SHAPES_IMPL_HPP_
+
+#include "cob_obstacle_distance/marker_shapes.hpp"
+
+/* BEGIN MarkerShape ********************************************************************************************/
+template <typename T>
+MarkerShape<T>::MarkerShape(T &fcl_object,
+      double x, double y, double z,
+      double quat_x, double quat_y, double quat_z, double quat_w,
+      double color_r, double color_g, double color_b, double color_a) : fcl_marker_converter_(fcl_object), is_drawn_(false)
```

Comment:
Already done with this ctor (line 68): 
MarkerShape(T &fcl_object, geometry_msgs::Point &pos, geometry_msgs::Quaternion &quat, std_msgs::ColorRGBA &col)


So one can sort what he or she wants.